### PR TITLE
Enable VREF and LPADC on FRDM_MCXW71

### PR DESCRIPTION
**Pull request by decsny**
MCXW71:
- VREF
- LPADC

and some other minor things regarding lpadc

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]
regulator changes ok

**Comment by decsny** [comment]
rebased

File: boards/nxp/frdm_mcxw71/doc/index.rst
Comment (danieldegrasse) [inline boards/nxp/frdm_mcxw71/doc/index.rst:1]: Title of this commit is:
> boards: frdm_mcxw71: Enable VREF and LPADC 

But it only touches the VREF support, so I think we need to update the message

File: soc/nxp/mcx/mcxw/soc.c [hunk 1]
```diff
@@ -119,6 +119,8 @@ static ALWAYS_INLINE void clock_init(void)
 	CLOCK_SetIpSrcDiv(kCLOCK_Lpi2c0, kSCG_SysClkDivBy16);
 	CLOCK_SetIpSrc(kCLOCK_Lpi2c1, kCLOCK_IpSrcFro192M);
 	CLOCK_SetIpSrcDiv(kCLOCK_Lpi2c1, kSCG_SysClkDivBy16);
+	CLOCK_SetIpSrc(kCLOCK_Lpadc0, kCLOCK_IpSrcFro192M);
```
Comment (danieldegrasse) [inline soc/nxp/mcx/mcxw/soc.c:122]: Can we keep this clocking within the `if (DT_NODE_HAS_COMPAT_STATUS(adc0, nxp_lpadc, okay))` block, or is it needed for multiple IP blocks?

**Review by danieldegrasse — COMMENTED** [review]
Just a few comments on the titles/content of commits

File: soc/nxp/mcx/mcxw/soc.c (hunk 1 above)
Comment (decsny) [reply soc/nxp/mcx/mcxw/soc.c:122]: I am just following the precedent in the file

**Review by danieldegrasse — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by dleach02** [comment]
@gmarull can you review

**Review by gmarull — DISMISSED** [review]
+1 for regulator changes

**Comment by fabiobaltieri** [comment]
heya, needs a rebase

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**